lib/HelloSignResource.js

```diff
@@ -331,11 +331,19 @@ HelloSignResource.prototype = {
             // submit as form
             var form = new FormData();
             for(var i=0; i<data.files.length;i++) {
-                form.append("file[" + i + "]", fs.createReadStream(data.files[i]));
+                if( typeof data.files[i] === 'string' ) {
+                    form.append("file[" + i + "]", fs.createReadStream(data.files[i]));
+                }
+                else if( data.files[i] instanceof Buffer ) {
+                    form.append("file[" + i + "]", data.files[i], {filename: 'signature.pdf'});
+                }
+                else if( data.files[i] instanceof Object &&
+                        data.files[i].hasOwnProperty('file') && data.files[i].file instanceof Buffer ) {
+                    form.append("file[" + i + "]", data.files[i].file, {filename: data.files[i].name||'signature.pdf'});
+                }
             }
-            delete data.files;
             for(var paramName in data){
-                if(data.hasOwnProperty(paramName)){
+                if(paramName != 'files' && data.hasOwnProperty(paramName)){
```

Comment:
Would you mind explaining why this change was required?

Reply:
Sure. The old code deleted the property "files" because it was just a string & once a file stream was opened the file name string was no longer needed. However, in the case of a buffer being passed in the "delete data.files" would delete the buffer. So, we just skip the 'files' property while serializing instead of deleting it first.